src/main/java/gui/CreationModalView.java

```diff
@@ -0,0 +1,215 @@
+import java.util.List;
+
+/**
+ * Class responsible for displaying a modal view for the creation of shots.
+ * @author alex
+ */
+public class CreationModalView extends ModalView {
+
+    private static final int width = 600;
+    private static final int height = 600;
+
+    private int numberOfCameras;
+
+    private VBox viewPane;
+    private List<CheckBox> cameraCheckboxes;
+    private TextField descripField;
+    private TextField nameField;
+    private NumberTextField startField;
+    private NumberTextField endField;
+    private Button creationButton;
+
+    private EventHandler<DirectorShotCreationEvent> shotCreationEventEventHandler;
+
+    /**
+     * Constructor.
+     * @param rootPane Pane to display modal on top of
+     * @param numberOfCamerasInTimeline Amount of cameras in timeline
+     * @param creationHandler Event handler for the creation of a shot
+     */
+    public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
+                             EventHandler<DirectorShotCreationEvent> creationHandler) {
+        super(rootPane, width, height);
+        this.numberOfCameras = numberOfCamerasInTimeline;
+        this.shotCreationEventEventHandler = creationHandler;
+        initializeCreationView();
+    }
+
+    /**
+     * Initialize and display the modal view.
+     */
+    private void initializeCreationView() {
+        this.viewPane = new VBox(20);
+        this.viewPane.getChildren().add(new Text("Add a new shot"));
+
+        initNameDescriptionFields();
+        initCountFields();
+        initCamCheckBoxes();
+
+        creationButton = new Button("Create");
+        creationButton.setOnMouseClicked(this::createShot);
+        this.viewPane.getChildren().add(creationButton);
+
+        super.setModalView(this.viewPane);
+        super.displayModal();
+    }
+
+    /**
+     * Initialize the shot name and description fields.
+     */
+    private void initNameDescriptionFields() {
+        // Add the textfields for shot's name & description
+        final Label nameLabel = new Label("Shot Name: ");
+        nameField = new TextField();
+        HBox nameBox = new HBox();
+        nameBox.getChildren().addAll(nameLabel, nameField);
+        nameBox.setSpacing(10);
+
+        final Label descripLabel = new Label("Shot Description: ");
+        descripField = new TextField();
+        HBox descripBox = new HBox();
+        descripBox.getChildren().addAll(descripLabel, descripField);
+        descripBox.setSpacing(10);
+
+        this.viewPane.getChildren().addAll(nameBox, descripBox);
+    }
+
+    /**
+     * Initialize the start and end count fields.
+     */
+    private void initCountFields() {
+        // Start and end points
+        final Label startLabel = new Label("Start:");
+        startField = new NumberTextField();
+        final Label endLabel = new Label("End:");
+        endField = new NumberTextField();
+
+        // Add default values as a cue
+        startField.setText("0");
```

Comment:
Maybe make these default values fields so we can edit them later (maybe on double click in the grid to add a block)